solidity/contracts/utility/ContractRegistryClient.sol

```diff
@@ -60,7 +61,7 @@ contract ContractRegistryClient is Owned, Utils {
         IContractRegistry newRegistry = IContractRegistry(addressOf(CONTRACT_REGISTRY));
 
         // verify that the new contract-registry is different and not zero
-        require(newRegistry != address(registry) && newRegistry != address(0), "ERR_INVALID_REGISTRY");
+        require(newRegistry != registry && address(newRegistry) != address(0), "ERR_INVALID_REGISTRY");

```

Comment:
Maybe `newRegistry != registry && newRegistry != IContractRegistry(0))`?

Reply:
don't think that's possible, see my comment above